primitives/src/parachain.rs

```diff
@@ -38,6 +38,9 @@ pub type CollatorSignature = ed25519::Signature;
 /// so we define it to be the same type as `SessionKey`. In the future it may have different crypto.
 pub type ValidatorId = super::SessionKey;
 
+/// Index of the validator is used as a lightweight replacement of the `ValidatorId` when appropriate.
+pub type ValidatorIndex = usize;
```

Comment:
`ValidatorIndex(usize)` will give more type-safety, but I'm not sure if it is important here.

Reply:
if it's in primitives, it should not be platform-dependent. u32 should be fine